### PR TITLE
shinken: init at 2.4.3

### PR DESCRIPTION
**Pull request by peterhoeg**
I want to add proper nixos module support for shinken which is going to be a lot of work, but the first step is to add the python module.

Because I don't really have a way to seed configuration data (the nixos module hasn't been written yet), this package is completely untested except for installation.
###### Things done:
- [X] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -9165,6 +9165,32 @@ in modules // {
...
+
+    # missing: logging tagging
+
+    propagatedBuildInputs = with self; [
+      arrow bottle cffi cherrypy django paramiko pycurl pymongo pyopenssl
+    ];
+
+    meta = {
+      homepage = http://www.shinken-monitoring.org/;
+      description = "Shinken Monitoring Framework";
```
Comment (jagajaga) [inline pkgs/top-level/python-packages.nix]: Please omit pkg's name. Use something like `A monitoring framework`.

Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: PyPI pages of packages often have a nice one-liner at the top. In this case *Shinken is a monitoring framework compatible with Nagios configuration and plugins*

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -9165,6 +9165,32 @@ in modules // {
     };
   };
 
+  shinken = buildPythonPackage rec {
+    version = "2.4.3";
+    name = "Shinken-${version}";
+    disabled = isPy3k;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: please include a comment explaining why

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -9165,6 +9165,32 @@ in modules // {
...
+
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/S/Shinken/${name}.tar.gz";
+      sha256 = "1hxbnxlk2bacb7p6lim9kqy50581m0d3p3y6b9hw0x800lfk0y6q";
+    };
+
+    dontStrip = true;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: please include a comment explaining why

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (jagajaga) [inline pkgs/top-level/python-packages.nix]: But anyway we omit pkg's name :)

**Comment by FRidh** [comment]
Looks good. Is `shinken` used only as application, or mainly as a Python library? In case of the former it is often desired to keep it outside of `python-packages.nix` since this is mostly for libraries although there is no strict separation. 

Did you test whether all tests run?

**Comment by peterhoeg** [comment]
```shinken``` is an application, not a library, so I will be moving it away from python-packages.nix.

I thought the installer would run tests directly, but it looks like there is a special script that needs to be run, so I will ensure that happens too.

Consider this wip. Thanks for your feedback @FRidh .

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -9165,6 +9165,32 @@ in modules // {
...
+
+    # missing: logging tagging
+
+    propagatedBuildInputs = with self; [
+      arrow bottle cffi cherrypy django paramiko pycurl pymongo pyopenssl
+    ];
+
+    meta = {
+      homepage = http://www.shinken-monitoring.org/;
+      description = "A monitoring framework compatible with Nagios configuration and plugins.";
```
Comment (jagajaga) [inline pkgs/top-level/python-packages.nix]: Please remove dot at the end.

**Comment by FRidh** [comment]
you will now need to call it from `all-packages.nix`

**Comment by FRidh** [comment]
see also https://github.com/FRidh/nixpkgs/tree/shinken

**Comment by peterhoeg** [comment]
I'll sort this out during the week. Haven't had time to look at it. There's quite a bit of code to be written to start using shinken on nixos.

**Comment by FRidh** [comment]
Just had a look at this. In setup.py there is a lot of custom code on where to install files to while most of this is normally taken care of by setuptools.

**Comment by peterhoeg** [comment]
@FRidh, the issue with shinken is that it tries to handle both install as well as various system admin functions on its own - it's really quite messy. I unfortunately haven't had any time to work on this (we are deploying on Debian at the moment which works OK) but will towards the end of June.

**Comment by FRidh** [comment]
This has been dormant for some time. Please reopen if it is ready to merge.

**Pull request closed without merging**